src/util/macros.rs

```diff
@@ -167,7 +167,7 @@ macro_rules! wait_op {
                  // we don't care error actually.
                 let _ = tx.send(res);
             };
-            $expr(cb);
+            $expr(cb)?;
```

Comment:
Please update the doc of the macro. It can only be used in functions return `Result` now.